bin/varnishd/cache/cache_hash.c

```diff
@@ -731,7 +731,12 @@ hsh_rush1(const struct worker *wrk, struct objcore *oc, struct rush *r)
 	Lck_AssertHeld(&oh->mtx);
 
 	AZ(oc->flags & OC_F_BUSY);
-	max = oc->flags ? cache_param->rush_exponent : INT_MAX;
+	if (oc->flags & OC_F_WITHDRAW)
```

Comment:
So we have now removed the rush limit from `HSH_DerefObjCore()` and replaced it with a new `HSH_Withdraw()` function and a new flag, to basically just implement what was the rush limit before.

How are we not just shuffling stuff from left to right here?

Reply:
I am driving this based on the objcore, so yes I'm shifting the rush policy from call sites to the objcore state, and no that's not just shuffling things around.

The "withdraw" logic should have been part of the "abandon" logic but I forgot about it by the time I "completed" the pull request. It was intended from the beginning though, as a couple commit message announce an increase of waiting list activity that should be temporary.

The good news I guess is that when I realized I had forgotten this detail I came up with "withdraw" that describes the use case better than "abandon" (and avoids overloading the latter).